eng/native/configurecompiler.cmake

```diff
@@ -606,6 +606,10 @@ if (CLR_CMAKE_HOST_UNIX)
     # other clang 16.0 suppressions
     add_compile_options(-Wno-single-bit-bitfield-constant-conversion)
     add_compile_options(-Wno-cast-function-type-strict)
+    
+    # clang 18.0 warns about VLAs being a C++ extension by default.
```

Comment:
Where do we use VLAs?

Reply:
There are two instance while building for Linux:
https://github.com/dotnet/runtime/blob/590d664b637db8ee6a95063f7927143d6ff66455/src/coreclr/gc/unix/gcenv.unix.cpp#L635
and
https://github.com/dotnet/runtime/blob/590d664b637db8ee6a95063f7927143d6ff66455/src/coreclr/pal/src/file/path.cpp#L402

Reply:
If it is just these two, I think it is better to convert them to use alloca.

Reply:
That should be easy, I'll change them tomorrow! Should I also check if there are more in code that's only used for different platforms? Also if there are no notes in the coding guidelines it might be worth mentioning that VLAs shouldn't be used (and maybe enable warnings for them?)

Reply:
> That should be easy, I'll change them tomorrow!

Thanks!

> Should I also check if there are more in code that's only used for different platforms?

I do not expect that you will find any. VLAs are not supported on Windows that prohibits their use in majority of the code. And there is very little code in the repo that is non-Windows non-Linux.

> there are no notes in the coding guidelines it might be worth mentioning that VLAs shouldn't be used (and maybe enable warnings for them

VLAs are very niche. I do not think it is worth mentioning them in the coding guidelines. Enabling the warnings for them on older compilers sounds good if it is possible.

Reply:
Converted them to ``alloca`` and enabled warnings for unix hosts. If I understand it correctly only MSVC is supported for Windows which doesn't support VLAs (for C++), for Linux there is Clang and GCC, both of which support ``-Wvla``, this should cover every supported compiler/platform.